tools/respecDocWriter.js

```diff
@@ -1,135 +1,173 @@
       const debugURL = `${url.origin}${url.pathname}`;
-      const msg = `${warn} ${colors.debug(debugURL)}`;
+      const msg = `📡 HTTP Error ${response.status()}: ${debugURL}`;
       throw new Error(msg);
     }
     log(`Navigation complete.`);
+
     await checkIfReSpec(page);
+    const version = await getVersion(page);
+    log(`Using ReSpec v${version.join(".")}`);
+
     log("Processing ReSpec document...");
-    const html = await generateHTML(page, url, timer);
+    const html = await generateHTML(page, timer, version, url);
     log("Processed document.");
-    const abortOnWarning = whenToHalt.haltOnWarn && haltFlags.warn;
-    const abortOnError = whenToHalt.haltOnError && haltFlags.error;
-    if (abortOnError || abortOnWarning) {
-      throw new Error(
-        `${abortOnError ? "Errors" : "Warnings"} found during processing.`
-      );
-    }
-    await write(out, html);
+
     // Race condition: Wait before page close for all console messages to be logged
     await new Promise(resolve => setTimeout(resolve, 1000));
     await page.close();
     log("Done.");
-    return html;
+
+    return { html, errors, warnings };
   } finally {
     await browser.close();
   }
 }
 
+/**
+ * Fetches a ReSpec "src" URL, and writes the processed static HTML to an "out" path.
+ * @deprecated Please use `convertToHTML` instead.
+ * @param {string} src A URL or filepath that is the ReSpec source.
+ * @param {string | null | ""} out A path to write to. If null, goes to stdout. If "", then don't write, just return value.
+ * @param {object} [whenToHalt] Allowing execution to stop without writing.
+ * @param {boolean} [whenToHalt.haltOnError] Do not write if a ReSpec processing has an error.
+ * @param {boolean} [whenToHalt.haltOnWarn] Do not write if a ReSpec processing has a warning.
+ * @param {object} [options]
+ * @param {number} [options.timeout] Milliseconds before processing should timeout.
+ * @param {boolean} [options.disableSandbox] See https://peter.sh/experiments/chromium-command-line-switches/#no-sandbox
+ * @param {boolean} [options.debug] Show the Chromium window with devtools open for debugging.
+ * @param {boolean} [options.verbose] Log processing status to stdout.
+ * @param {(error: RsError) => void} [options.onError] What to do if a ReSpec processing has an error. Logs to stderr by default.
+ * @param {(warning: RsError) => void} [options.onWarning] What to do if a ReSpec processing has a warning. Logs to stderr by default.
+ * @return {Promise<string>} Resolves with HTML when done writing. Rejects on errors.
+ */
+async function fetchAndWrite(src, out, whenToHalt = {}, options = {}) {
+  const colors = require("colors");
+  colors.setTheme({ debug: "cyan", error: "red", warn: "yellow" });
+
+  console.warn(
+    colors.warn(
+      "DEPRECATION WARNING: `fetchAndWrite` is deprecated and will be removed in a future version. Please use `convertToHTML` instead."
+    )
+  );
```

Comment:
I wonder if we should should define generic `showWarn(msg)` and `showError(msg)`. Would avoid the ugly looking:

```
console.error(colors.error(
```